crates/wysiwyg/src/composer_model/delete_text.rs

```diff
@@ -71,8 +72,24 @@ where
     /// Deletes the character after the current cursor position.
     pub fn delete(&mut self) -> ComposerUpdate<S> {
         if self.state.start == self.state.end {
-            // Go forward 1 from the current location
-            self.state.end += 1;
+            let mut next_char_len: isize = 1;
+            let (s, e) = self.safe_selection();
+            let range = self.state.dom.find_range(s, e);
+            let leaves: Vec<&DomLocation> = range.leaves().collect();
+            if s == e && leaves.len() == 1 {
+                let leaf = leaves[0];
+                if let DomNode::Text(text_node) =
+                    self.state.dom.lookup_node(&leaf.node_handle)
```

Comment:
This chunk of code is repeated between here and do_backspace - I think it should be in a shared function.

Reply:
It's also a bit awkward - we actually just want "which text node is at this location?" - maybe we should write a function to answer that question. (Bear in mind sometimes there is no text node - e.g. when the cursor is between 2 BRs)